modules/drivers/bigquery-cloud-sdk/src/metabase/driver/bigquery_cloud_sdk.clj

```diff
@@ -323,10 +323,7 @@
       (let [db-id (u/the-id database)]
         (log/infof (trs "DB {0} had hardcoded dataset-id; changing to an inclusion pattern and updating table schemas"
                         db-id))
-        (doseq [table (db/select MetabaseTable :db_id db-id)]
-          (let [table-id (u/the-id table)]
-            (log/infof (trs "Updating table {0} to set schema to dataset-id of {1}" table-id dataset-id))
-            (db/update! MetabaseTable table-id :schema dataset-id)))
+        (db/update-where! MetabaseTable {:db_id db-id} :schema dataset-id)
```

Comment:
```suggestion
        (db/update-where! MetabaseTable {:db_id db-id, :schema nil} :schema dataset-id)
```

Reply:
Just to be extra safe and only update things where `schema IS NULL`

Reply:
That's a change to what the previous code would do. Seems to make sense but would need @jeff303 to chime in with what assumptions are there.

Reply:
```suggestion
        (db/execute! {:update MetabaseTable
                      :set    {:schema dataset-id}
                      :where  [:and
                               [:= :db_id db-id]
                               [:or
                                [:= :schema nil]
                                [:not= :schema dataset-id]]]})
```